### DISCUSSION TITLE
pitching amplitude increment UDF build problem in SU2

### DISCUSSION THREAD

**Discussion by HANSIeltsKing** [su2code/SU2#1729] [Q&A]
I am an SU2 user simulating config_incomp_turb_sa.cfg. I would like to define a UDF to simulate an airfoil vibration dynamics.
the original configuration feature is:
```
GRID_MOVEMENT= RIGID_MOTION
MOTION_ORIGIN= (0.25 0.0 0.0)
PITCHING_OMEGA= (0.0 0.0 14.91675)
PITCHING_AMPL= (0.0 0.0 8.0)
```
The pitching amplitude is a fixed number. however, in my simulation task, i would like to build a UDF to show an increasing pitching amplitude, which depends on the value of the current timestep. I would like to know if there is a related testcase that shows how a UDF is written. Thank you very much.

**Comment by pcarruscag** [comment]
I don't think we'll ever have "input UDF's" since we are open source you can just go in and code what you want. I'm converting this to discussion, feel free to ask about where to implement this in the code there.
